src/scripts/medgen_conflicts_removals.py

```diff
@@ -0,0 +1,119 @@
+  https://github.com/monarch-initiative/mondo-ingest/issues/273#issuecomment-1632774012
+"""
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import List, Set, Union
+
+import pandas as pd
+
+SCRIPTS_DIR = Path(__file__).parent
+PROJECT_DIR = SCRIPTS_DIR.parent.parent
+ONTOLOGY_DIR = PROJECT_DIR / "src" / "ontology"
+TMP_DIR = ONTOLOGY_DIR / "tmp"
+INPUT_CONFLICTS = str(TMP_DIR / "July2023_CUIReports_FromMedGentoMondo.xlsx")
+INPUT_OBO = str(ONTOLOGY_DIR / "mondo-edit.obo")
+OUTPUT_FILE = str(TMP_DIR / "mondo-edit.obo.tmp")
+
+
+def collate_xref_removals(input_file: str = INPUT_CONFLICTS) -> List[str]:
+    """Collate xref removals"""
+    book = pd.ExcelFile(input_file)
+    bad_xref_ids: Set[Union[str, int]] = set()
+
+    # Sheet 1/5: Bad_CNxRefs_withCUI
+    # - summary: MedGen team has suggested us better mappings to use on these Mondo IDs.
+    # - recommended action: For Mondo terms identified in "mondo_id", remove the xrefs identified in "mondo_xref_bad".
+    df = book.parse("Bad_CNxRefs_withCUI")
+    bad_xref_ids.update(df['mondo_xref_bad'].tolist())
+
+    # Sheet 2/5: WrongCUIxref_withCurrCUI
+    # - summary: MedGen team has suggested us better mappings to use on these Mondo IDs.
+    # - recommended action: For Mondo terms identified in "mondo_id", remove the xrefs identified in "mondo_xref_bad".
+    df = book.parse("WrongCUIxref_withCurrCUI")
+    bad_xref_ids.update(df['mondo_xref_bad'].tolist())
+
+    # Sheet 3/5: 1MondoID_to1CUI
+    # - summary: MedGen has identified better Mondo IDs to map to these CUIs.
+    # - recommended action: For terms in the "mondo_id(Mondo)" column, remove xrefs identified in the "UMLS_CUI" column.
+    df = book.parse(">1MondoID_to1CUI")
+    bad_xref_ids.update(df['UMLS_CUI'].tolist())
+
+    # Sheet 4/5: Mondo_MedGen_CUImismatch_1Mondo
+    # - summary: We have Mondo terms that have xrefs to multiple MedGen CUIs. Only 1 of them should be left in.
+    # - recommended action: Remove incorrect xrefs; the ones not identified in the "MedGenCUI" column.
+    df = book.parse("Mondo_MedGen_CUImismatch_1Mondo")
+    to_remove = list(df.apply(
+        lambda row: [x for x in row['medgen_query'].split(' OR ')
+                     if 'MONDO' not in x
+                     and x != row['MedGenCUI']][0], axis=1))
+    bad_xref_ids.update(to_remove)
+
+    # Sheet 5/5: MondoXrefs_NotinMedGen
+    # - summary: These terms do not exist in MedGen.
+    # - recommended action: Remove these xrefs.
+    df = book.parse("MondoXrefs_NotinMedGen")
+    bad_xref_ids.update(df['ref_target'].tolist())
+
+    # Convert to strings and sort
+    bad_xref_ids: List[str] = sorted([str(x) for x in bad_xref_ids])
+    return bad_xref_ids
+
+
+def run(input_conflicts: str = INPUT_CONFLICTS, input_obo: str = INPUT_OBO, output: str = OUTPUT_FILE):
+    """Run"""
+    # Vars
+    exceptions = ['MONDO:preferredExternal', 'https://orcid.org']
```

Comment:
##  Removal exceptions `MONDO:preferredExternal` & `https://orcid.org`
@matentzn Update 8/31: We just met today and I'm re-looking at the comments we didn't look over. This is a copy/paste of my [original comment](https://github.com/monarch-initiative/mondo/pull/6560#discussion_r1306588345).

The main point:
Is that you told me not to remove cases where I see these strings. But the only time they come up is when Megan specifically is saying that I should remove the xref. I'm assuming you want me to follow Megan's instructions in these cases.

---

You said:
> If a row contains one of the following strings, do not remove it: MONDO:preferredExternal, https://orcid.org (indicates someone has manually removed it)

I'm only doing 3 kinds of deletes. Here they are, as well as a count of how many times I found either of these strings:
1. source="MEDGEN: - 0
2. xref: MEDGEN: - 0
3. Medgan's "xref: UMLS:" conflicts - 23

So, when it comes to my removals of anything MEDGEN/UMLS related, the only time I'm finding these strings (in the xref axiom annotations) is for cases where Megan specifically asked me to remove an xref.

However, I take it that you do not actually want me to override her, so (as can be seen in my Python snippet above), I did not implement this exception logic for those conflicts.

Also, just to clarify--what do you mean by "it" in "indicates someone has manually removed it" in your quote above? I think it means "UMLS or MEDGEN source= evidence axiom annotations". Is that correct?

Reply:
Do you know without much work how often the case happens that a removed xref has an orcid? Dont do a lot work, I can see it from the diff as well.

Reply:
If it's useful info, in the 'details' above, I wrote:
> I'm only doing 3 kinds of deletes. Here they are, as well as a count of how many times I found either of these strings:
> source="MEDGEN: - 0
> xref: MEDGEN: - 0
> Medgan's "xref: UMLS:" conflicts - 23

As for _orcid_ specifically, looks like just 2 cases:
`git diff master src/ontology/mondo-edit.obo | grep orcid | grep -- -xref`
```
-xref: UMLS:C1512224 {source="DOID:5310", source="MONDO:equivalentTo", source="MONDO:notFoundInDiseaseSubset", source="https://orcid.org/0000-0001-5493-2602"}
-xref: UMLS:C4755316 {source="MONDO:equivalentTo", source="https://orcid.org/0000-0001-5493-2602"}
```